### PR TITLE
Add disk device gathering customized ansible-module

### PR DESCRIPTION
**Pull request by tonytan4ever**

### CODE REVIEW

File: playbooks/maas-host-cdm.yml [hunk 1]
```diff
@@ -48,6 +48,14 @@
         - deploy_osp | bool
 
   tasks:
+    - name: Discover disk device facts
+      gather_disk_device_facts:
+        hostname: "{{ ansible_hostname }}"
+        container_name: "{{ container_name | default(inventory_hostname) }}"
```
Comment (cloudnull) [inline playbooks/maas-host-cdm.yml]: will this module return anything? if so, do we need a "register" ?

File: playbooks/library/gather_disk_device_facts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    def __init__(self, module):
+        self.state_change = False
+        self.module = module
+
+    def gather_facts(self, hostname, container_name):
+        """Get disk device facts on a host"""
+        # in case we need to gather disks inside of a container
+        _ = container_name
+        _ = _ = hostname
+        disk_device_command_string = ""
```
Comment (cloudnull) [inline playbooks/library/gather_disk_device_facts]: If Im reading this right, this is an empty command so it'd return nothing? Is this supposed to be user configurable, or is that coming soon?

File: playbooks/maas-host-cdm.yml (hunk 1 above)
Comment (tonytan4ever) [reply playbooks/maas-host-cdm.yml]: The module technically won't return anything, but set a fact on the host running it:

https://github.com/rcbops/rpc-maas/pull/617/files#diff-6cd182564ec970218f86b4c5d2626820R60

**Comment by tonytan4ever** [comment]
recheck_bionic_mnaio_osp13_deploy

**Comment by tonytan4ever** [comment]
Looks like the [getdevinfo](https://gitlab.com/hamishmb/getdevinfo) package does not work well.  Switch to [pyudev](https://pyudev.readthedocs.io/en/latest/) instead.

**Comment by tonytan4ever** [comment]
recheck_all

**Comment by tonytan4ever** [comment]
recheck_bionic_mnaio_osp13_deploy

**Comment by tonytan4ever** [comment]
recheck_bionic_mnaio_osp13_deploy

**Comment by tonytan4ever** [comment]
recheck_bionic_mnaio_osp13_deploy

**Comment by tonytan4ever** [comment]
recheck_bionic_mnaio_osp13_deploy

**Comment by tonytan4ever** [comment]
recheck_xenial_queens_deploy
recheck_bionic_mnaio_osp13_deploy

**Comment by tonytan4ever** [comment]
recheck_xenial_pike_deploy

**Comment by tonytan4ever** [comment]
recheck_xenial_pike_deploy

**Comment by tonytan4ever** [comment]
recheck_bionic_mnaio_osp13_deploy

**Pull request merged**